library/core/src/slice/iter.rs

```diff
@@ -3423,3 +3430,12 @@ impl<'a, T: 'a + fmt::Debug, P> fmt::Debug for ChunkByMut<'a, T, P> {
         f.debug_struct("ChunkByMut").field("slice", &self.slice).finish()
     }
 }
+
+/// Same as `p.addr().get()`, but faster to compile by avoiding a bunch of
+/// intermediate steps and unneeded UB checks, which also inlines better.
+#[inline]
+fn addr_usize<T>(p: NonNull<T>) -> usize {
```

Comment:
Is there a case where the existence of this method can be made unnecessary? It feels weird that the alternative would perform so poorly, or involve UB checks at all.

Reply:
Hmm, I'm mostly just wondering if this is sufficient motivation to add this method on `NonNull` directly (it can be `pub(crate)` for now), whether this hack *is* in fact specific to this module, or whether this function should be marked as FIXME and should be addressed later.